### DISCUSSION TITLE
AVX2 benefits

### DISCUSSION THREAD

**Discussion by tfbryan** [datalust/seq-tickets#1450] [General]
We recently became aware that our Seq servers did not have the AVX2 instruction set enabled. Is this setting generally expected to be enabled or it is something uniquely utilized by Seq? 

Now that we have it enabled, is there anything we can do to measure the benefit it's providing?

**Comment by nblumhardt** [comment]
Hi Todd,

Great to hear! AVX2 is very widely used, it's a vectorized instruction set supported by most modern x64 CPUs.

Seq uses AVX2 to accelerate JSON parsing deep in its storage subsystem. We published a write-up just a little while ago at: https://blog.datalust.co/deserializing-json-really-fast/

The performance gain will be reduced CPU usage when pulling historical data from the disk archive. This might not be apparent in wall clock time in situations where IO is the bottleneck, but load on the machine should be eased by enabling AVX2.

Hope this helps,
Nick
